### ISSUE TITLE
handler error not finished upload

### ISSUE THREAD

**Issue by jcsena** [expressjs/multer#225]
handler error not finished upload 

``` javascript
     var multerConf = multer({
       dest: config.directory ,
       fileFilter:fileFilter
     });

     app.post('/api/v2/uploads/ths',  multerConf.single('file'),function(req,res){
       res.status(200).send({file: req.file});
     });

     app.use(function (err, req, res, next) {
       res.status(413).send('File too large');
     });

     function fileFilter (req, file, cb) {
      cb(new Error('Empty file'));
    }
```
when ``` cb(new Error('Empty file'))```  send error but not finished upload multipart , and  when i use ``` cb(null,false); ``` send ok with req.file empty and finished event upload multipart

how is the correct form for send the errors to view?

**Comment by LinusU** [comment]
This should be fixed by #205 and released on npm as version `1.0.5` :rocket: 

Please reopen if issue persists.

-----

I believe that you did the right thing but that there was a bug preventing it from working, please try with `1.0.5` :+1:

**Comment by jcsena** [comment]
the bug is persistent, when I start and first request multer this catch errors ```{ [Error: File too large] code: 'LIMIT_FILE_SIZE', field: 'file', storageErrors: [] } ``` , but the next request the error is ```undefined```  

``` javascript

//config
  var multerUpload =  multer({
      dest: config.directory,
      limits: {
        fileSize: 1 * 1024 * 1024,
        files: 1
      }
    })

    exports.upload = function(req,res,next){

      return multerUpload(req, res, function (err) {

        console.log(err);
          if(err) return res.status(400).json(err);
          next();
          // Everything went fine
        });

    };

```
